src/fs/bare/tree.ts

```diff
@@ -0,0 +1,101 @@
+import basic from '../network/basic'
+import { Link, Links, SimpleTree, SimpleFile } from '../types'
+import check from '../types/check'
+import { CID, FileContent } from '../../ipfs'
+import { constructors as BareFileConstructors } from '../bare/file'
+import BaseTree from '../base/tree'
+import { removeKeyFromObj } from '../../common'
+import link from '../link'
+import semver from '../semver'
+
+class BareTree extends BaseTree {
+
+  links: Links
+
+  constructor(links: Links) {
+    super(semver.v0)
+    this.links = links
+  }
+
+  async createEmptyTree(): Promise<BareTree> {
+    return constructors.empty()
+  }
+
+  async createTreeFromCID(cid: CID): Promise<BareTree> {
+    return constructors.fromCID(cid)
+  }
+
+  async createFile(content: FileContent): Promise<SimpleFile> {
+    return BareFileConstructors.create(content)
+  }
+
+  async createFileFromCID(cid: CID): Promise<SimpleFile> {
+    return BareFileConstructors.fromCID(cid)
+  }
+
+  async put(): Promise<CID> {
+    return basic.putLinks(this.links, null)
+  }
+
+  async updateDirectChild(child: SimpleTree | SimpleFile, name: string): Promise<this> {
```

Comment:
I introduced `SimpleTree` as an interface for a `BareTree` (ie "pretty tree"). which has slightly more limited capabilities than a `Tree` which is assumed to have a `Header`. There are a few ways around this:
- Remove the common parent (which is `BaseTree` right now). This removes a layer of abstraction and I don't think we'd need the separate interface anymore. But we would also have to duplicate a lot of the code that is shared beween `BareTree`s and `Public`/`PrivateTrees`
- hamfist `BareTree` information into a `Header` (basically add filler info, etc) so that they both fulfill the `Tree` interface. This was the approach I did before this refactor and it seemed to get pretty confusing.
- change `SimpleTree` to `Tree` and `Tree` to `HeaderTree`: this doesn't change the fact that we have two interfaces, but it might be better semantics?

If you have other thoughts I'm happy to hear them! I wasn't too happy about adding a second interface, but I didn't think the other tradeoffs were worth it

Reply:
I like the idea of `Tree` and `HeaderTree`, since Tree/SimpleTree is used so much.

Reply:
Maybe sticking a little README in `src/fs/` with a overview of all the types of trees could help. If you have time.